### PR TITLE
Fix boost/any.hpp transitive dep

### PR DESCRIPTION
**Pull request by bryanloz-xilinx**
#### Problem solved by the commit
XRT Native APIs are using boost/any.hpp in their public interface. Yet, the CMake infrastructure does not inform consumers that they must use the boost include directory in the project include path.

### CODE REVIEW

**Comment by bryanloz-xilinx** [comment]
Ongoing discussion with @stsoe , please don't merge immediately

File: src/runtime_src/core/pcie/linux/CMakeLists.txt [hunk 1]
```diff
@@ -55,6 +55,8 @@ target_link_libraries(xrt_core
   rt
   dl
   uuid
+  INTERFACE
+  Boost::boost
```
Comment (stsoe) [inline src/runtime_src/core/pcie/linux/CMakeLists.txt:59]: Why this change for xrt_core?   I follow the xrt_coreutil because that library is required for xrt::device which compile time depends on boost.  But xrt_core library is not linked directly (usually) to host application and even if it was, it doesn't have client requirements beyond the run-time dependencies that are installed automatically with the rpm/deb.

File: src/runtime_src/core/pcie/linux/CMakeLists.txt [hunk 2]
```diff
@@ -71,6 +73,7 @@ target_link_libraries(xrt_core_static
   dl
   rt
   pthread
+  Boost::boost
```
Comment (stsoe) [inline src/runtime_src/core/pcie/linux/CMakeLists.txt:76]: ditto.

**Review by stsoe — COMMENTED** [review]
Thanks Bryan for taking a stab at this.  Please check my question,

File: src/runtime_src/core/pcie/linux/CMakeLists.txt (hunk 1 above)
Comment (bryanloz-xilinx) [reply src/runtime_src/core/pcie/linux/CMakeLists.txt:59]: We are linking both of them
```
$ ldd rt-engine/librt-engine.so 
        linux-vdso.so.1 (0x00007ffe1fdfb000)
        libxrt_core.so.2 => /opt/xilinx/xrt/lib/libxrt_core.so.2 (0x00007f8f901da000)
        libxrt_coreutil.so.2 => /opt/xilinx/xrt/lib/libxrt_coreutil.so.2 (0x00007f8f8feab000)
```

I guess I didn't know which library I should apply this to, so I chose both.

Comment (bryanloz-xilinx) [reply src/runtime_src/core/pcie/linux/CMakeLists.txt:59]: Also, what about hardware emulation? In that case I'm also not sure which library to apply this to.

Comment (stsoe) [inline src/runtime_src/core/pcie/linux/CMakeLists.txt:59]: Let's remove from libxrt_core.

Comment (stsoe) [reply src/runtime_src/core/pcie/linux/CMakeLists.txt:59]: hw_emu is same as xrt_core, it should not need the dependency.

File: src/runtime_src/core/pcie/linux/CMakeLists.txt (hunk 2 above)
Comment (stsoe) [reply src/runtime_src/core/pcie/linux/CMakeLists.txt:76]: I hear you about std::any, and for sure we'd prefer that, unfortunately C++17 is not yet universally supported on all the platforms we need to build against.  But maybe we should consider some conditionally magic, at least that would resolve it for your use-case and we can then punt on the INTERFACE changes?

**Comment by rozumx** [comment]
> Ongoing discussion with @stsoe , please don't merge immediately

Bryan, I have added the DO NOT MERGE tag.  When you are ready to have these changes merged, please remove it.

File: src/runtime_src/core/pcie/linux/CMakeLists.txt (hunk 2 above)
Comment (rozumx) [reply src/runtime_src/core/pcie/linux/CMakeLists.txt:76]: Hmm...If there are C++ 17 issues, shouldn't the XRT-Pipeline catch them for it is my understanding that each supported platform is built.

**Review by rozumx — APPROVED** [review]

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by stsoe** [comment]
@bryanloz-xilinx I will be closing this PR and adding another one that default switches to c++17 and std::any.

**Comment by bryanloz-xilinx** [comment]
> @bryanloz-xilinx I will be closing this PR and adding another one that default switches to c++17 and std::any.

Sounds good @stsoe
